Sticky posts are acting weird in front page template

PBruk opened the pull request:
Tagging @frrrances for code-review: Checks for sticky posts, returns nothing otherwise. Resolves #54.

PBruk committed:
Checks for sticky posts, returns nothing otherwise.

frrrances commented:
Looks good but my php is rusty... tagging @matt-bernhardt for a second review.

matt-bernhardt commented:
The changes to inc/content-front.php seem to introduce a syntax error that's being flagged by CodeSniffer - specifcially, there's what looks like the tail end of an if/else structure on lines 38-42, but I don't see it opening anywhere. 

My suspicion is that if I deploy this branch, it will crash the front page of any site that uses that template.

My assumption is that there needs to be a WP_Query on this template that grabs all (and only) sticky posts for a loop?

matt-bernhardt committed:
Fixes incorrectly nested if/while loops

The most important change here is to implement a check for the length of
$sticky. Wordpress, if asked to query a list of zero posts, will default
to returning everything apparently?

matt-bernhardt commented:
Some notes on this re-written branch...

The approach started by @PBruk was valid, assuming that WP_Query behaves in a sane way. Unfortunately, in this case that doesn't seem to be true. 

The original approach was to generate an array of sticky posts using `get_option( 'sticky_posts' )`, and then sort and slice it to five items. That sorted array was then fed to a WP_Query object, which worked great so long as the array had _any_ items. Unfortunately, it appears that WP_Query, when presented with an array of zero length, instead returns _all_ items.

Because of this, I've refactored the page slightly, to check for the length of the array generated by `get_option( 'sticky_posts' )`. If that array has zero length, then the entire section gets skipped - which cuts down on a bit of computation and avoids the irrational need of WP_Query to return _something_.

With that, we're ready for a new code review by one or both of @PBruk and @frrrances

frrrances commented on inc/content-front.php:
```diff
@@ -15,27 +15,46 @@
+
+				the_content();
+
+				if ( is_active_sidebar( 'sidebar-two' ) ) : ?>
```
why the double sidebar?

matt-bernhardt replied:
Line 57 isn't new, and is part of an existing check for the sidebar to make sure it is rendered only when needed. The original construct was

```
                <?php if ( is_active_sidebar( 'sidebar-two' ) ) : ?>

                    <?php dynamic_sidebar( 'sidebar-two' ); ?>

                <?php endif; ?>     
```

Looking at this, I actually didn't go quite far enough - it is somewhat illegible IMO to be closing and opening PHP tags unnecessarily - so what might be better is:

```
                if ( is_active_sidebar( 'sidebar-two' ) ) :

                    dynamic_sidebar( 'sidebar-two' );

                endif;

```

frrrances replied:
Ah I see now. And I agree with your cleanup ideas!

PBruk replied:
Looks good to me!

2016-06-29 14:15 GMT-04:00 frances botsford notifications@github.com:

> In inc/content-front.php
> https://github.com/MITLibraries/MITLibraries-child/pull/59#discussion_r68999355
> :
> 
> > -                           <div class="excerpt-post__body">
> > -                               <h3><a href="<?php echo the_permalink();?>"><?php echo get_the_title();?></a></h3>
> > -                               <?php custom_excerpt( 20, '...' ) ?>
> > -                           </div>
> > -                       </div>
> >   +
> > -                       <?php endwhile;
> >   +
> > -                       wp_reset_postdata();
> >   +
> > -                   } // End "have posts".
> > -               } // End "if no sticky posts".
> >   +
> > -               the_content();
> >   +
> > -               if ( is_active_sidebar( 'sidebar-two' ) ) : ?>
> 
> Ah I see now. And I agree with your cleanup ideas!
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> https://github.com/MITLibraries/MITLibraries-child/pull/59/files/991cea9f73635d15db20466cb0031cc94e09d2ac#r68999355,
> or mute the thread
> https://github.com/notifications/unsubscribe/ABv1VYwaNXvF8GtM9XvtzaAv5DeQiei7ks5qQrZfgaJpZM4I8CLt
> .

## 

"The reason is never the reason."

_\- Some French Movie Director_"Every portrait that is painted with feeling
is a portrait of the artist, not the sitter."
_\- Oscar Wilde_

frrrances commented:
Overall, makes sense to me, though i'll let @PBruk review the php details. I had one small comment at the end as it looked like the sidebar would be rendered twice.

matt-bernhardt commented:
This is going to collide with the fix for #65, which has also made changes to `content-front.php`. I'm not sure which should merge first, but let's discuss tomorrow.

frrrances commented:
👍
